diagnostic_feedback/tests/unit/test_diagnostic_feedback.py

```diff
@@ -0,0 +1,298 @@
+import json
+import os
+from base_test import BaseTest
+from .wizard_step_mixin import WizardStepMixin
+
+from nose.tools import (assert_equals)
+
+
+class DiagnosticFeedbackAjaxTest(BaseTest, WizardStepMixin):
+    _oldMaxDiff = None
+
+    ZONE_1 = None
+    ZONE_2 = None
+
+    FEEDBACK = {
+        0: {"correct": None, "incorrect": None},
+        1: {"correct": None, "incorrect": None},
+        2: {"correct": None, "incorrect": None}
+    }
+
+    FINAL_FEEDBACK = None
+
+    def __init__(self, *args, **kwargs):
+        self._step1_data = None
+        self._step2_data = None
+        self._step3_buzzfeed_data = None
+        self._step3_diagnostic_data = None
+        self._block = None
+        super(DiagnosticFeedbackAjaxTest, self).__init__(*args, **kwargs)
+
+    @classmethod
+    def setUpClass(cls):
+        cls._oldMaxDiff = assert_equals.__self__.maxDiff
+        assert_equals.__self__.maxDiff = None
+
+    @classmethod
+    def tearDownClass(cls):
+        assert_equals.__self__.maxDiff = cls._oldMaxDiff
+
+    def setUp(self):
+        self._block = self.make_block()
+        self._step1_data = self.load_step_data(1)
+        self._step2_data = self.load_step_data(2)
+        self._step3_buzzfeed_data = self.load_step_data(3, self._block.BUZZFEED_QUIZ_VALUE)
+        self._step3_diagnostic_data = self.load_step_data(3, self._block.DIAGNOSTIC_QUIZ_VALUE)
+
+    def tearDown(self):
+        self._block = None
+
+    def load_json_resource(self, path):
+        return open(os.path.join(os.path.dirname(__file__), '..', path)).read()
+
+    def load_step_data(self, step, quiz_type=None):
+        if step == 1:
+            return json.loads(self.load_json_resource('data/step1_test_data.json'))
+        elif step == 2:
+            return json.loads(self.load_json_resource('data/step2_test_data.json'))
+        elif step == 3:
+            if quiz_type == self._block.BUZZFEED_QUIZ_VALUE:
+                return json.loads(self.load_json_resource('data/step3_buzzfeed_test_data.json'))
+            else:
+                return json.loads(self.load_json_resource('data/step3_diagnostic_test_data.json'))
+
+    def test_wizard_step1(self):
+
+        for _type, data in self._step1_data.items():
+            data = json.dumps(data)
+            res = json.loads(self._block.handle('save_data', self.make_request(data)).body)
+
+            if _type == 'missing_step':
+                assert_equals(res['success'], False)
+            elif _type == 'missing_title':
+                assert_equals(res['success'], False)
+            elif _type == 'invalid_type':
+                assert_equals(res['success'], False)
+            elif _type == 'missing_title_type':
+                assert_equals(res['success'], False)
+            elif _type == "missing_description":
+                assert_equals(res['success'], False)
+            elif _type == 'valid_data':
+                assert_equals(res['success'], True)
```

Comment:
This test passes if `_type` is set to something other than the options given here.  I assume that's not intentional. Consider an `else: fail()` block, or an explicit check like `assert_in(_type`, `set(['missing_step', 'missing_title', 'invalid_type', 'missing_title_type', 'missing_description', 'valid_data']))`